app/models/spree/user.rb

```diff
@@ -132,16 +132,6 @@ def default_card
       end
     end
 
-    def generate_spree_api_key!
-      self.spree_api_key = SecureRandom.hex(24)
-      save!
-    end
```

Comment:
I reckon that the generation of the API key is part of the data structure and we can keep that in the model. I would probably replace this with a simpler version to be more flexible:
```rb
    def generate_api_key
      self.spree_api_key = SecureRandom.hex(24)
    end
```

Then the controller can call that method and then call `save` separately.